core/src/ast_builder/table.rs

```diff
@@ -9,20 +13,32 @@ use super::AlterTableNode;
 use super::{CreateIndexNode, DropIndexNode, OrderByExprNode};
 
 #[derive(Clone)]
-pub struct TableNode {
+pub enum TableType<'a> {
+    Table,
+    Series(ExprNode<'a>),
+    Dictionary(Dictionary),
+    Derived {
+        subquery: Box<SelectNode<'a>>,
+        alias: String,
+    },
+}
+
+#[derive(Clone)]
+pub struct TableNode<'a> {
     pub table_name: String,
+    pub table_type: TableType<'a>,
 }
 
-impl TableNode {
-    pub fn alias_as(self, table_alias: &str) -> TableAliasNode {
+impl<'a> TableNode<'a> {
+    pub fn alias_as(self, table_alias: &'a str) -> TableAliasNode {
```

Comment:
```suggestion
    pub fn alias_as(self, table_alias: &str) -> TableAliasNode<'a> {
```

I think it is correct to declare this lifetime in TableAliasNode, not in str.